### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by atok624**
> # Ruby on Rails capstone project - Budget app
> In this pull request, the following tasks were implemented:
> 
> * Splash screen
>   
>   * A simple page with the name of your app, and links to the signup and log-in pages.
> * Sign up and login pages
>   
>   * The user should be able to register in the app with full name, email, and password.
>   * The user can log into the app using email and password.
>   * If the user is not logged in, they can't access pages that require the user to be logged in.
> * Home page (categories page)
>   
>   * When the user logs in, they are presented with the categories page.
>   * For each category, the user can see their name, icon, and the total amount of all the transactions that belong to that category.
>   * When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
>   * There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
> * Transactions page
>   
>   * The list of transactions is presented for a given category, ordered by the most recent.
>   * At the top of the page, the user could see the total amount for the category (the sum of all of the amounts of the transactions in that category).
>   * There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
>   * When the user clicks on the "Back" button (<), the user navigates to the home page.
> * Add a new category" page
>   
>   * The user fills out a form to create a new category, indicating their name and icon (both mandatory).
>   * The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
>   * When the user clicks on the "Back" button (<), the user navigates to the home page.
> * Add a new transaction" page
>   
>   * The user fills out a form to create a new transaction with:
>     name (mandatory)
>     amount (mandatory)
> Categories (mandatory at least one)
>   * The user clicks (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
>   * When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
> * Create unit and integration tests for all the most important components of your RoR application.

### CODE REVIEW

File: app/views/payments/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    <% else %>
+  <% @payments.each do |payment| %>
+  <div class="single-payment">
+      <p class="payment-name"><%= payment.name %></p>
+      <div class="flex">
+        <p>Price: $ <%= payment.amount %> </p>
+        <%# <i class=" flex1 fa-solid fa-trash"></i> %>
+      </div>
+    </div>
+  <% end %>
```
Comment (elisha2kyakpo1) [inline app/views/payments/index.html.erb]: - [ ] Kindly ensure that your design matches the given design, currently, the transactions and the categories pages background color are different, let's fix this.

design


- [ ] Great job adding this page to list all the transactions, unfortunately, the total amount of transactions should be calculated and recorded at the top.

total-tr

File: app/views/payments/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    </a>
+</div>
+
+<%= form_with(model: @payment, url: category_payments_path) do |form| %>
+   
+<div class="form-container">
+        <%= form.text_field :name, placeholder: "Name" %>
+        <%= form.text_field :amount, placeholder: "Amount" %>
+         <%= form.submit "Create", class: "submit-button" %>
+    </div>
```
Comment (elisha2kyakpo1) [inline app/views/payments/new.html.erb]: - [ ] Kindly add another field for category selection (check box or dropdown menu, etc.), this is part of the project requirements.

selection

File: app/views/categories/_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    <a href="/logout">
+        <i class="fa-solid fa-right-from-bracket"></i>
+    </a>
+</div>
+<%= form_with(model: @category)do |form| %>
+    <div class="form-container">
+        <%= form.text_field :name, placeholder: "Name" %>
+        <%= form.text_field :icon, placeholder: "Icon Link" %>
+         <%= form.submit "Create", class: "submit-button" %>
+    </div>
```
Comment (elisha2kyakpo1) [inline app/views/categories/_form.html.erb:16]: - [ ] Kindly fix the submit button for the categories, the requirement is that when a user clicks the submit button, the app navigates to the home page.

selection

File: config/routes.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+Rails.application.routes.draw do
+  resources :categories do
+    resources :payments
```
Comment (elisha2kyakpo1) [inline config/routes.rb]: - [ ] Kindly expose the routes in use such as index, show, new, etc, except put/update.

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @atok624, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description are good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.
✔️ Tests passing👏

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/payments/new.html.erb (hunk 2 above)
Comment (atok624) [reply app/views/payments/new.html.erb]: Resoved

File: app/views/categories/_form.html.erb (hunk 3 above)
Comment (atok624) [reply app/views/categories/_form.html.erb:16]: Resolve

File: config/routes.rb (hunk 4 above)
Comment (atok624) [reply config/routes.rb]: Resolved

**Review by anagudelogu — APPROVED** [review]
# STATUS: APPROVED! 🍾 🎆 🥇 

Hi @atok624, 
I'm Andres.

You did a great job working on this milestone, your project is complete! 🚀 
There is nothing else to say other than... it's time to merge it 🤖  
Congratulations! 🎉 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Great UI design. 💻 
✅ Descriptive PR title, summary and commit messages. 💬 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by atok624** [comment]
Thanks so much for the review @anagudelogu

**Pull request merged**